src/fsearch_window.c

```diff
@@ -725,6 +725,62 @@ on_listview_row_is_selected(int row, gpointer user_data) {
     return FALSE;
 }
 
+struct GListAccumulator {
+    GList *list;
+};
+typedef struct GListAccumulator GListAccumulator;
+
+static void
+map_to_uri_helper(gpointer key, gpointer value, gpointer user_data) {
+    FsearchDatabaseEntry *entry = value;
+    if (G_UNLIKELY(!entry)) {
+        return;
+    }
+
+    GString *path = db_entry_get_path_full(entry);
+    if (G_UNLIKELY(!path)) {
+        g_warning("[map_to_uri_helper] Failed to retrieve path for entry at idx %d", db_entry_get_idx(entry));
+        return;
+    }
+
+    gchar *uri = g_filename_to_uri(path->str, NULL, NULL);
+    if (G_UNLIKELY(!uri)) {
+        g_warning("[map_to_uri_helper] Failed to convert path to URI: %s", path->str);
+        return;
+    }
+
+    GListAccumulator *acc = user_data;
+    acc->list = g_list_prepend(acc->list, uri);
+
+    g_string_free(path, TRUE);
+}
+
+static gchar **
+listview_on_drag_data_get(gpointer user_data) {
+    FsearchApplicationWindow *win = FSEARCH_APPLICATION_WINDOW(user_data);
+    FsearchDatabaseView *db_view = win->result_view->database_view;
+    if (!db_view) {
+        g_warning("[listview_on_drag_data_get] Database view is NULL.");
+        return NULL;
+    }
+
+    GListAccumulator acc = { NULL };
+    db_view_selection_for_each(db_view, map_to_uri_helper, &acc);
+
+    gchar **uri_array = g_new(gchar *, g_list_length(acc.list) + 1);
+    int i = 0;
+    for (GList *l = g_list_last(acc.list); l != NULL; l = l->prev) {
```

Comment:
I suppose you're walking the list from the last to the first element to get the order right? That's fine in theory, but just be aware that currently `db_view_selection_for_each()` has no defined order in which it walks through the selected entries, as the selection is stored in a hash table. So the order will likely be messed up either way.

Reply:
Still, the code is fine as it is. We really should at least try to get the order right, in case we switch data structures for selections.